//
// Copyright 2019 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo.sgx;

// A CPUSVN contains security-relevant information about the Intel microcode
// version and running authenticated code modules.
message CpuSvn {
  // The bytes of the CPUSVN. Must be exactly 16 bytes long. Required.
  optional bytes value = 1;
}

// The possible values for an SGX Type extension in a PCK certificate.
//
// WARNING: The tag numbers do NOT correspond to the ASN.1 ENUMERATED integer
// values in PCK certificates.
enum SgxType {
  // `SGX_TYPE_UNKNOWN` does not represent a value possible in a PCK
  // certificate. It exists so that the default value for an `SgxType` cannot be
  // confused for a real SGX Type.
  SGX_TYPE_UNKNOWN = 0;
  STANDARD = 1001;
}

// Security-relevant machine configuration properties.
message MachineConfiguration {
  optional CpuSvn cpu_svn = 1;    // Optional.
  optional SgxType sgx_type = 2;  // Optional.
}

// Specification of which fields from MachineConfiguration to match.
message MachineConfigurationMatchSpec {
  // Flag indicating whether to perform a match on |cpu_svn|. Required.
  optional bool is_cpu_svn_match_required = 1;

  // Flag indicating whether to perform a match on |sgx_type|. Required.
  optional bool is_sgx_type_match_required = 2;
}
